packages/react-native/ReactCommon/react/nativemodule/core/platform/android/ReactCommon/JavaTurboModule.cpp

```diff
@@ -391,37 +411,115 @@ jsi::Value createJSRuntimeError(
       .call(runtime, message);
 }
 
+/**
+ * Cpp struct to hold JVMFrame
+ */
+struct SimpleJFrame {
+    std::string className;
+    std::string fileName;
+    int lineNumber;
+    std::string methodName;
+};
+
+/**
+ * Cpp struct to hold JThrowable
+ */
+struct SimpleJThrowable {
+    std::string name;
+    std::string message;
+    std::vector<SimpleJFrame> stackElements;
+};
+
+/**
+ * Extracts throwable properties to JVM env independent data holder.
+ */
+SimpleJThrowable extractThrowable(
+            jni::local_ref<jni::JThrowable> throwable) {
+    SimpleJThrowable extracted;
+    auto stackTrace = throwable->getStackTrace();
+
+    extracted.stackElements.reserve(stackTrace->size());
+    for (int i = 0; i < stackTrace->size(); ++i) {
+        auto frame = stackTrace->getElement(i);
+        extracted.stackElements.push_back(SimpleJFrame {
+                frame->getClassName(),
+                frame->getFileName(),
+                frame->getLineNumber(),
+                frame->getMethodName()
+        });
+    }
+
+    extracted.name = throwable->getClass()->getCanonicalName()->toStdString();
+    extracted.message = throwable->getMessage()->toStdString();
+
+    return extracted;
+}
+
+/**
+ * Converts simple JVM independent Throwable container to JSError.
+ */
+jsi::JSError convertSimpleJThrowableToJSError(
+        jsi::Runtime& runtime,
+        SimpleJThrowable throwable) {
+    auto stackTrace = throwable.stackElements;
```

Comment:
Grab a reference to avoid a copy
```suggestion
    auto &stackTrace = throwable.stackElements;
```